src/aiida/storage/sqlite_dos/backend.py

```diff
@@ -208,3 +209,19 @@ def nodes(self):
     @cached_property
     def users(self):
         return orm.SqliteUserCollection(self)
+
+    @staticmethod
+    @lru_cache(maxsize=18)
+    def _get_mapper_from_entity(entity_type: 'EntityTypes', with_pk: bool):
+        """Return the Sqlalchemy mapper and fields corresponding to the given entity.
+
+        :param with_pk: if True, the fields returned will include the primary key
+        """
+        from sqlalchemy import inspect
+
+        from ..sqlite_zip.models import MAP_ENTITY_TYPE_TO_MODEL
```

Comment:
feels quite weird that we import from a different backend. Should there perhaps be `sqlite_common` or something like that? (maybe add a TODO here)

Reply:
A refactoring might be useful but not necessary right now. Was planning on doing this sometime in the future. There is also some duplication between `psql_dos` and `sqlite_dos` both using the DOS repository backend. Basically `sqlite_dos` is a bit of a Frankenstein monster taking bits from other storage plugins ^^